mobile_index/src/pending_txn.rs

```diff
@@ -0,0 +1,190 @@
+use crate::{Error, Result};
+use chrono::{DateTime, Utc};
+use futures::TryFutureExt;
+use serde::{Deserialize, Serialize};
+
+#[derive(sqlx::Type, Serialize, Deserialize, Debug)]
+#[sqlx(type_name = "status", rename_all = "lowercase")]
+pub enum Status {
+    Created,
+    Pending,
+    Cleared,
+    Failed,
+}
+
+impl Status {
+    fn update_query(&self) -> &'static str {
+        match self {
+            Self::Created => {
+                r#" 
+                update pending_txn set 
+                    status = $1
+                where hash = $2; 
+                "#
+            }
+            Self::Pending => {
+                r#" 
+                update pending_txn set 
+                    status = $1, 
+                    submitted_at = $3 
+                where hash = $2; 
+                "#
+            }
+            Self::Cleared | Self::Failed => {
+                r#" 
+                update pending_txn set 
+                    status = $1, 
+                    completed_at = $3 
+                where hash = $2; 
+                "#
+            }
+        }
+    }
+
+    fn update_all_query(&self) -> &'static str {
+        match self {
+            Self::Created => {
+                r#" 
+                update pending_txn set 
+                    status = $1
+                where hash = any($3); 
+                "#
+            }
+            Self::Pending => {
+                r#" 
+                update pending_txn set 
+                    status = $1, 
+                    submitted_at = $2 
+                where hash = any($3); 
+                "#
+            }
+            Self::Cleared | Self::Failed => {
+                r#" 
+                update pending_txn set 
+                    status = $1, 
+                    completed_at = $2
+                where hash = any($3); 
+                "#
+            }
+        }
+    }
+}
+
+#[derive(sqlx::FromRow, Deserialize, Serialize, Debug)]
+#[sqlx(type_name = "pending_txn")]
```

Comment:
What even is a pending txn anymore on solana when you’re adding to the lifetime balance of a hotspot? Shouldn’t it either update the db immediately or else throw an error since the indexer is just adding the monotonically increasing reward total for each hotspot? And in the mobile world shouldn’t this be attached to the mobile radio instead of the hotspot since someone could sell or give away their radio but the lifetime rewards for that radio would stay with the hotspot right now?

Reply:
Yup,, copy pasta from mobile_rewards. Cleared out a bunch of files now just to get a clean baseline while we do the config switchover. I'll fix up the schema for mobile_index as soon as that lands